22.2.4+ (#1)
friedcucumber/maui-infinite-grid-scrolling: README.md

<!-- default badges list -->
<!-- default badges end -->
# DataGrid for .NET MAUI - Infinite Scrolling

> This example demonstrates how to implement infinite scrolling in [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid).

Author: ```suggestion
This example demonstrates how to implement infinite scrolling in the [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid).
```

> This example demonstrates how to implement infinite scrolling in [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid).

Reviewer: Because I can interpret "Data Grid" as a proper noun. Moreover, we've got a link to the corresponding documentation article here.

<img width="30%" src="https://user-images.githubusercontent.com/12169834/230103799-e52c384f-3efe-4428-b663-35756da1fe59.png"/>

## Implementation Details

> * Set the [IsLoadMoreEnabled](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.IsLoadMoreEnabled) to `true` to enable the Data Grid infinite scrolling.

Reviewer: First. We do not use articles with proper nouns. I would have added the word "property" after the property name to resolve this issue.

Second. We do not use a definite article together with abstract/mass nouns unless absolutely necessary. I speak of the word "scrolling" here.

```suggestion
* Set the [IsLoadMoreEnabled](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.IsLoadMoreEnabled) property to `true` to enable infinite scrolling in Data Grid.
```

> * Bind the [DataGridView.LoadMoreCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.LoadMoreCommand) property to a view model's command that loads new items in the command's Execute method. Data Grid automatically invokes this command when a user scrolls to the end.

Reviewer: > Bind the [DataGridView.LoadMoreCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.LoadMoreCommand) property to a view model's command that loads new items in the command's Execute method.

This fragment is really strange.

> a view model's **command that loads new lines in the command's Execute method**.

Do we speak of the same command in both cases? If we do, then something is redundant/amiss here. I recommend you to ponder on this sentence and try to rewrite it from scratch.

Thanks in advance.

Author: ```suggestion
* Bind the [DataGridView.LoadMoreCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.LoadMoreCommand) property to a view model's command that loads new items. Data Grid automatically invokes this command when a user scrolls to the end.
```

What about this one?

Reviewer: I see. However, I see one more issue here:

> when a user **scrolls** to the end.

First question -- scrolls what? I suppose we can say "content" here.
Second question -- the end? Of what end do we speak here in this "infinite scrolling" scenario? Maybe the last loaded item/entry?

Right now, I see this sentence as follows:

```suggestion
* Bind the [DataGridView.LoadMoreCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.LoadMoreCommand) property to a view model's command that loads new items. Data Grid automatically invokes this command when a user scrolls content to the last loaded item.
```

What do you think? Does this change make sense to you?

> * Set the [LoadMoreCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.LoadMoreCommand)'s **CanExecute** property to `false` to disable infinite scrolling when you scrolled down to the end of DataGrid's data.

Reviewer: > when you scrolled down to the end of DataGrid's data.

"DataGrid's data" does not sound good. But, more importantly, what all this means, in simpler words?

It means that the scroll position reaches the last record in Data Grid -- the way I see it.

```suggestion
* Set the [LoadMoreCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.LoadMoreCommand)'s **CanExecute** property to `false` to disable infinite when the scroll position reaches the last record in Data Grid.
```

> * While loading a new portion of data, you can set the [IsRefreshing](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.IsRefreshing) property to `true` to display wait indicator. When the loading is over, set [IsRefreshing](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.IsRefreshing) to `false` to automatically update the DataGridView's displayed data.

Reviewer: Indicator is countable, therefore, requires an article. An indefinite article, in this context, because we only introduce it. I also moved the initial clause to the end of the first sentence to break the pattern when two sentences in a row start with a clause.

> automatically

What does this word give us in this context? We can safely remove it because our action already updates data (once we set IsRefreshing to false).

> **the** DataGridView's displayed data.

Data is an abstract/uncountable/mass noun. Just displayed data. Also, I don't believe we should mention `DataGridView` here because we are already have all proper context, there will be no confusion.

As a result, we end up with the following:

```suggestion
* You can set the [IsRefreshing](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.IsRefreshing) property to `true` to display a wait indicator while Data Grid is loading a new portion of data. Once the load operation is complete, set [IsRefreshing](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.IsRefreshing) to `false` to update displayed data.
```

> * To load only the next portion of data, you can store a number of displayed rows and pass it to the [Skip](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.skip) method. Then, pass the number of rows that you want to display in DataGrid to the [Take](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.take) method.

Reviewer: > the number of rows **that** you want to display

We can safely omit "that", and nothing's gonna change. If we can accomplish our task with fewer words, we should.

```suggestion
* To load only the next portion of data, you can store a number of displayed rows and pass it to the [Skip](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.skip) method. Then, pass the number of rows you want to display in DataGrid to the [Take](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.take) method.
```

    ```
    allBlogs.Skip(startIndex).Take(batchSize);
    ```

    File to look at: [MainViewModel.cs](CS/MainViewModel.cs)

> * Call the [Browser.OpenAsync](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/appmodel/open-browser?view=net-maui-7.0&tabs=android#open-the-browser) and [Share.RequestAsync](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/data/share?view=net-maui-7.0&tabs=android#share-text-and-links) methods to open the blog post in browser and display the native share dialog.

Reviewer: First. We do not need a definite article when we list named entities, according to the new guideline on articles.

Second. **The** blog post? Of what blog post do we speak? We don't have it defined anywhere. If we speak of our blog post, we should've included a link to it.

Third. Browser is countable; therefore, an article is necessary. If we want to use a definite article, we need to define this browser somehow. For example, the default browser.

Right now, I see this paragraph in the following form:

```suggestion
* Call [Browser.OpenAsync](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/appmodel/open-browser?view=net-maui-7.0&tabs=android#open-the-browser) and [Share.RequestAsync](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/data/share?view=net-maui-7.0&tabs=android#share-text-and-links) methods to open a blog post in a browser and display the native share dialog.
```

However, we should clarify the question regarding this strange blog post.

Author: This sample includes a blog post list that a user can scroll infinitely.

Reviewer: OK. Thank you for the clarification. Then looks like everything is fine.
